v2/hack/certify/certify.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+set -Eeox pipefail
+
+SLEEP_LONG="${SLEEP_LONG:-5}"
+SLEEP_SHORT="${SLEEP_SHORT:-2}"
+CERT_NAMESPACE="${CERT_NAMESPACE:-rhm-certification}"
+KUBECONFIG="${KUBECONFIG:-$HOME/.kube/config}"
+
+# Check Subscriptions: subscription-name, namespace
+checksub () {
+	echo "Waiting for Subscription $1 InstallPlan to complete."
+
+	# Wait 2 resync periods for OLM to emit new installplan
+	sleep 60
+
+	# Wait for the InstallPlan to be generated and available on status
+	unset INSTALL_PLAN
+	until oc get subscription $1 -n $2 --output=jsonpath={.status.installPlanRef.name}
+	do
+		sleep $SLEEP_SHORT
+	done
+
+	# Get the InstallPlan
+	until [ -n "$INSTALL_PLAN" ]
+	do
+		sleep $SLEEP_SHORT
+		INSTALL_PLAN=$(oc get subscription $1 -n $2 --output=jsonpath={.status.installPlanRef.name})
+	done
+
+	# Wait for the InstallPlan to Complete
+	unset PHASE
+	until [ "$PHASE" == "Complete" ]
+	do
+		PHASE=$(oc get installplan $INSTALL_PLAN -n $2 --output=jsonpath={.status.phase})
+    if [ "$PHASE" == "Failed" ]; then
+      set +x
+      sleep 3
+      echo "InstallPlan $INSTALL_PLAN for subscription $1 failed."
+      echo "To investigate the reason of the InstallPlan failure run:"
+      echo "oc describe installplan $INSTALL_PLAN -n $2"
```

Comment:
Nice comment!

Reply:
checksub routine was written back in pre-ICPA days, before there was a product and ansible installer, and OLM was a little rougher around the edges

this at least got people to not immediately ask me blindly why a hacked up pre-prod install didn't work...

Reply:
Hah - it's the little things tho.